session/session_test.go

```diff
@@ -3316,61 +3315,106 @@ PARTITION BY RANGE (c) (
+	tk.MustExec("insert into t1 (c) values (101)") // write dc-2 with global scope
+	result = tk.MustQuery("select * from t1")      // read dc-1 and dc-2 with global scope
+	c.Assert(len(result.Rows()), Equals, 3)
+	c.Assert(txn.Valid(), IsTrue)
+	tk.MustExec("rollback")
 	result = tk.MustQuery("select * from t1")
+	c.Assert(len(result.Rows()), Equals, 2)
+
+	tk.MustExec("insert into t1 (c) values (101)") // write dc-2 with global scope
+	result = tk.MustQuery("select * from t1")      // read dc-1 and dc-2 with global scope
 	c.Assert(len(result.Rows()), Equals, 3)
 
 	// set txn_scope to local
 	tk.MustExec("set @@session.txn_scope = 'dc-1';")
 	result = tk.MustQuery("select @@txn_scope;")
 	result.Check(testkit.Rows("dc-1"))
-	// test local txn
-	tk.MustExec("insert into t1 (c) values (1)") // in dc-1 with dc-1 scope
-	result = tk.MustQuery("select * from t1 where c < 100")
+
+	// test local txn auto commit
+	tk.MustExec("insert into t1 (c) values (1)")            // write dc-1 with dc-1 scope
+	result = tk.MustQuery("select * from t1 where c < 100") // read dc-1 with dc-1 scope
 	c.Assert(len(result.Rows()), Equals, 3)
+
+	// begin and commit with dc-1 txn scope
 	tk.MustExec("begin")
 	txn, err = tk.Se.Txn(true)
 	c.Assert(err, IsNil)
 	c.Assert(tk.Se.GetSessionVars().TxnCtx.TxnScope, Equals, "dc-1")
 	c.Assert(txn.Valid(), IsTrue)
-	tk.MustExec("insert into t1 (c) values (1)") // in dc-1 with dc-1 scope
-	result = tk.MustQuery("select * from t1 where c < 100")
+	tk.MustExec("insert into t1 (c) values (1)")            // write dc-1 with dc-1 scope
+	result = tk.MustQuery("select * from t1 where c < 100") // read dc-1 with dc-1 scope
 	c.Assert(len(result.Rows()), Equals, 4)
 	c.Assert(txn.Valid(), IsTrue)
 	tk.MustExec("commit")
 	result = tk.MustQuery("select * from t1 where c < 100")
 	c.Assert(len(result.Rows()), Equals, 4)
 
-	// test wrong scope local txn
-	_, err = tk.Exec("insert into t1 (c) values (101)") // in dc-2 with dc-1 scope
-	c.Assert(err.Error(), Matches, ".*out of txn_scope.*")
+	// begin and rollback with dc-1 txn scope
+	tk.MustExec("begin")
+	txn, err = tk.Se.Txn(true)
+	c.Assert(err, IsNil)
+	c.Assert(tk.Se.GetSessionVars().TxnCtx.TxnScope, Equals, "dc-1")
+	c.Assert(txn.Valid(), IsTrue)
+	tk.MustExec("insert into t1 (c) values (1)")            // write dc-1 with dc-1 scope
+	result = tk.MustQuery("select * from t1 where c < 100") // read dc-1 with dc-1 scope
+	c.Assert(len(result.Rows()), Equals, 5)
+	c.Assert(txn.Valid(), IsTrue)
+	tk.MustExec("rollback")
 	result = tk.MustQuery("select * from t1 where c < 100")
 	c.Assert(len(result.Rows()), Equals, 4)
+
+	// test wrong scope local txn auto commit
+	_, err = tk.Exec("insert into t1 (c) values (101)") // write dc-2 with dc-1 scope
+	c.Assert(err.Error(), Matches, ".*out of txn_scope.*")
+	err = tk.ExecToErr("select * from t1 where c > 100") // read dc-2 with dc-1 scope
+	c.Assert(err.Error(), Matches, ".*can not be read by.*")
+
+	// begin and commit with dc-1 txn scope
```

Comment:
There is already an identical comment `// begin and commit with dc-1 txn scope` at L3366, maybe a new comment emphasizing this commit is failed?